tests/Bridge/OpenAI/Embeddings/ResponseConverterTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Tests\Bridge\OpenAI\Embeddings;
+
+use PhpLlm\LlmChain\Bridge\OpenAI\Embeddings\ResponseConverter;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Small;
+use PHPUnit\Framework\Attributes\Test;
+use PHPUnit\Framework\TestCase;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+#[CoversClass(ResponseConverter::class)]
+#[Small]
+class ResponseConverterTest extends TestCase
+{
+    #[Test]
+    public function itConvertsAResponseToAVectorResponse(): void
+    {
+        $response = $this->createStub(ResponseInterface::class);
```

Comment:
Why not using a JsonMockResponse directly?

Reply:
Do you maybe have more information to get me on your level here? I am a bit stuck with that you could mean with a `JsonMockResponse` 🤔

Reply:
![CleanShot 2024-12-06 at 12 49 02@2x](https://github.com/user-attachments/assets/c93a1550-fc27-4221-b228-ad2765255f6a)

Reply:
i think in this case the benefit is not really there yet - more like a question of style, e.g. not using mock frameworks but implementations.

in general i can think of some infrastructure for MockResponse from file where also processing is done already - when oop design is done, it could be a quite common scenario to test those response converters and contract is `ResponseInterface` yup